File: open_close/1.cs
public class Rectangle: Shape
{
    public double Width;
    public double Height;
}

public class Circle: Shpae{
    public double Radius;
}



public double Area(object[] shapes)
{
    double area = 0;
    foreach (var shape in shapes)
    {
        if (shape is Rectangle)
        {
            Rectangle rectangle = (Rectangle) shape;
            area += rectangle.Width*rectangle.Height;
        }
        else
        {
            Circle circle = (Circle)shape;
            area += circle.Radius * circle.Radius * Math.PI;
        }
    }

    return area;
}


public class AreaCalculator
{
    public double Area(Rectangle[] shapes)
    {
        double area = 0;
        foreach (var shape in shapes)
        {
            area += shape.Width*shape.Height;
        }

        return area;
    }
}

File: liskov/1.java
class Rectangle{
	void setWidth(double width){
		this.width = width;
	}
	void setHeight(double height){
		this.height = height;
	}

	double getArea(){

		return height*width;
	}
}


// Square is a rectangle

class Square extends Rectangle{

	void setHeight(double height){
		this.width = height;
		this.height = height;
	}

	void setWidth(double width){
		this.width = width;
		this.height = width;
	}
}